gnosis-buyback/contracts/Buyback.sol

```diff
@@ -0,0 +1,539 @@
+            _sender.transfer(_tipAmount);
+            emit Tip(_userAddress, _sender, _tipAmount);
+        }
+    }
+
+    /**
+     * @notice burnTokens
+     */
+    function burnTokens(address _tokenAddress, address _burnAddress, uint _amount) internal {
+        // transfer the tokens to address(0)
+        require(_amount > 0);
+        require(Token(_tokenAddress).transfer(_burnAddress, _amount));
+        emit Burn(
+            _tokenAddress,
+            address(0),
+            _amount
+        );
+    }
+
+    /**
+     * @notice withdraw
+     * @param _tokenAddress Contract address of token to withdraw
+     * @param _toAddress Address to send token to
+     * @param _amount Amount of tokens to withdraw
+     */
+    function withdraw(
+        address _tokenAddress, 
+        address _toAddress, 
+        uint _amount
+        ) external {
+            
+        address _userAddress = msg.sender;
+        uint userBalance = balances[_userAddress][_tokenAddress];
+        // current user auction total
+        uint userAuctionTotal = userBuybackAmountTotal[_userAddress][_tokenAddress];
+        require(_amount <= userBalance, "amount exceeds available balance");
+        // user isn't allowed to withdraw amount that affects
+        // the buyback
+        require(userAuctionTotal.add(_amount) <= userBalance, "user not allowed to withdraw balance affecting buyback");
+        require(_amount > 0, "withdrawal amount is not greater than zero");
+
+
+        require(userBalance >= _amount, "user balance is less than withdrawal amount");
+
+        userBalance = userBalance.sub(_amount);
+        balances[_userAddress][_tokenAddress] = userBalance; // set new balance
+
+        require(Token(_tokenAddress).transfer(_toAddress, _amount));
+
+        emit Withdraw(_userAddress, _tokenAddress, _amount, userBalance);
+    }
+
+    /**
+     * @notice withdrawEther
+     * @param _toAddress Address to send token to
+     * @param _amount Amount of tokens to withdraw
+     */
+    function withdrawEther(
+        address payable _toAddress, 
+        uint _amount
+        ) external {
+
+        address userAddress = msg.sender;
+        uint userBalance = etherBalance[userAddress];
+        // current tip auction total
+        uint userTipTotal = userTipAmountTotal[userAddress];
+        
+        require(_amount > 0, "withdrawal amount is not greater than zero");
+        require(userTipTotal.add(_amount) <= userBalance, "user balance is less than available withdrawal amount");
+        // withdrawing ether does not affect the current buybacks
+
+        userBalance = userBalance.sub(_amount);
+        etherBalance[userAddress] = userBalance; // set new balance
+        _toAddress.transfer(_amount);
+        emit WithdrawEther(userAddress, _amount, userBalance);
+    }
+
+    /**
+     * @notice removeBuyBack
+     * Allows releasing expired buyback funds
+     */
+    function releaseBuyBackFund(uint _buybackId) public {
```

Comment:
For me, what it make sense here is to release the auctions that has passed.

- You know the current auction for the token pair
- You can release the amounts for past auctions if no-one triggered the `postSellOrder`

Reply:
The only issue here is when a user specifies 0 as the auctionIndex it becomes impossible to 
releaseFunds for that buyback